src/app/chat/views/reactions.nim

```diff
@@ -42,17 +48,20 @@ QtObject:
         return pair[0]
     return ""
 
-  proc toggle*(self: ReactionView, messageId: string, emojiId: int) {.slot.} =
-    let emojiReactionId = self.messageEmojiReactionId(self.activeChannel.id, messageId, emojiId)
+  proc toggle*(self: ReactionView, messageId: string, chatId: string, emojiId: int) {.slot.} =
+    let emojiReactionId = self.messageEmojiReactionId(chatId, messageId, emojiId)
     if (emojiReactionId == ""):
-      self.status.chat.addEmojiReaction(self.activeChannel.id, messageId, emojiId)
+      self.status.chat.addEmojiReaction(chatId, messageId, emojiId)
     else:
       self.status.chat.removeEmojiReaction(emojiReactionId)
     
   proc push*(self: ReactionView, reactions: var seq[Reaction]) =
     let t = reactions.len
     for reaction in reactions.mitems:
-      let messageList = self.messageList[][reaction.chatId]
+      let chat = self.status.chat.channels[reaction.chatId]
+      var messageList = self.messageList[][reaction.chatId]
+      if chat.chatType == ChatType.Profile:
+        messageList = self.messageList[][status_utils.getTimelineChatId()]
```

Comment:
Same way, when new reactions are coming in, we need to put them in the timeline message list, in case they are coming from a profile update